src/mlpack/methods/random_forest/random_forest_impl.hpp

```diff
@@ -507,6 +507,22 @@ double RandomForest<
   #pragma omp parallel for reduction( + : totalGain)
   for (size_t i = 0; i < numTrees; ++i)
   {
+    // NOTE: this is a hacky workaround for older versions of Armadillo that did
+    // not (by default) set a different seed for each RNG.  We simply manually
+    // set the RNG seed for each individual thread.  However, if users have
+    // specifically set the RNG seed for Armadillo, this could break their code.
+    // So we hide it behind an (undocumented except for here) ifndef so that
+    // this support can be disabled if desired...
+    #if (ARMA_VERSION_MAJOR < 12) || \
+        ((ARMA_VERSION_MAJOR == 12) && (ARMA_VERSION_MINOR < 6) && \
+         (ARMA_VERSION_PATCH < 2))
```

Comment:
Not sure this version check would work reliably.  For example, the workaround is not activated if we have Armadillo version 12.4.2, which is affected by the RNG seed bug.

Suggest to adapt the following approach, which removes ambiguities:

````
#undef  MLPACK_ARMA_VERSION
#define MLPACK_ARMA_VERSION (ARMA_VERSION_MAJOR * 10000 + ARMA_VERSION_MINOR * 100 + ARMA_VERSION_PATCH)
  
#if (MLPACK_ARMA_VERSION < 120602)
  ... activate workaround ...
#endif
````

Reply:
Ahhh, this is what happens when writing code while also in a meeting.  Will fix shortly...